### ISSUE TITLE
Error reporting from within Cockpit

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#19386]
Currently, Cockpit does not have error reporting.

The closest we have is "Ooops!" which asks people to open up developer tools and copy and paste the message there to a GitHub issue (which we don't even mention).

We do not provide any way for people to report issues to get them fixed. We hope that people will figure out that the "web console" is known as "Cockpit" and that it would be reported here on GitHub. Further, we have some pages like Machines that have a different place to go to report the issue. (The split is probably arbitrary from the point of view of someone using Cockpit.)

Anaconda has added bug reporting; we might want to consider using something like it. 

Additionally, we could even construct the body text with system information that gets sent to GitHub as a query string, with the current page, system information, and a short description. For example: https://github.com/cockpit-project/cockpit/issues/new?labels=bug,page:metrics&title=test&body=stuff (opens a new issue with a label of "page:metrics" and "bug"). However, this requires a GitHub account. (But a GitHub account is needed to file a bug, regardless.)

**Comment by garrett** [comment]
Additionally, for an "Ooops!", we require people to open the dev tools... and console messages are quite weird to those who don't know what the message mean:

![image](https://github.com/cockpit-project/cockpit/assets/10246/00a2c0de-3a4d-4166-bd81-fdc2a014b872)

**Comment by martinpitt** [comment]
Agreed that the current "Oops" handling is too thin. It should at least bring up a dialog with the oops message and a copy&paste button, plus some references where to check.

However, I wonder if we should really build in the straight Oops → GitHub new issue link. GitHub does not have a builtin duplication detection the way e.g. Bugzilla or Launchpad have. So it could very easily end up in a situation where e.g. Fedora 39 has some web UI crash which got fixed upstream ages ago, but nevertheless users report the same thing over and over again as new bugs.

Experience in my previous life with [Apport](https://launchpad.net/apport): that did have an automated and elaborate client-side duplication, and still Launchpad got swamped with crash reports. Of course that scale was blewn up as a looooot more users are e.g. running LibreOffice or running `apt upgrade` than users who run the Fedora installer. But the same consideration applies in principle.

Perhaps we could do some middle ground and link to Anaconda's bugzilla and/or GitHub issue tracker, so that users can at least check for an existing bug reports first before they file a new one?

@jkonecny12 @M4rtinK, @KKoukiou any thoughts?

**Comment by martinpitt** [comment]
Let's close this. Our GitHub issues are already an unmaintainable graveyard, we *really* don't need even more with lesser quality.
